influxdb/src/main/java/com/hazelcast/jet/influxdb/InfluxDbSources.java

```diff
@@ -0,0 +1,283 @@
+     * Authenticates with the server using given credentials.
+     *
+     * @param query             query to execute on InfluxDb database
+     * @param database          name of the database
+     * @param url               url of the InfluxDb server
+     * @param username          username of the InfluxDb server
+     * @param password          password of the InfluxDb server
+     * @param chunkSize         the number of {@link QueryResult}s to process in one chunk.
+     * @param measurementMapper mapper function which takes measurement name, tags set, column names and row
+     *                          as argument and produces the user object {@link T} which will be emitted from
+     *                          this source
+     * @param <T>               type of the user object
+     * @return {@link T} mapped user objects
+     */
+    @Nonnull
+    public static <T> BatchSource<T> influxDb(@Nonnull String query, @Nonnull String database, @Nonnull String url,
+                                               @Nonnull String username, @Nonnull String password, int chunkSize,
+                                               @Nonnull QuadFunction<String, Map<String, String>,
+                                                       List<String>, List<Object>, T> measurementMapper) {
+        checkTrue(query != null, "query cannot be null");
+        checkTrue(url != null, "url cannot be null");
+        checkTrue(database != null, "database cannot be null");
+        checkTrue(username != null, "username cannot be null");
+        checkTrue(password != null, "password cannot be null");
+        checkTrue(measurementMapper != null, "measurementMapper cannot be null");
+
+        return influxDb(query, database, () -> connect(url, username, password).setDatabase(database), chunkSize,
+                measurementMapper);
+    }
+
+    /**
+     * Creates a source that executes the query on given database and
+     * emits items mapped with user defined mapper function.
+     * Uses given {@link InfluxDB} instance to interact with the server.
+     *
+     * @param query              query to execute on InfluxDb database
+     * @param database           name of the database
+     * @param connectionSupplier supplier which returns {@link InfluxDB} instance
+     * @param chunkSize          the number of {@link QueryResult}s to process in one chunk.
+     * @param measurementMapper  mapper function which takes measurement name, tags set, column names and row
+     *                           as argument and produces the user object {@link T} which will be emitted from
+     *                           this source
+     * @param <T>                type of the user object
+     * @return {@link T} mapped user objects
+     */
+    @Nonnull
+    public static <T> BatchSource<T> influxDb(@Nonnull String query, @Nonnull String database,
+                                              @Nonnull SupplierEx<InfluxDB> connectionSupplier, int chunkSize,
```

Comment:
we shouldn't require database name is user is already supplying the connection, it's this way in the sink.